bindings/rust/src/lib.rs

```diff
@@ -289,6 +289,74 @@ impl TypedExecutionResult {
 }
 
 impl Instance {
+    /// Ensure the range is valid according to the currently available memory size.
+    fn checked_memory_range(
+        &self,
+        offset: u32,
+        size: usize,
+    ) -> Result<core::ops::Range<usize>, ()> {
+        let offset = offset as usize;
+        let memory_size = self.memory_size();
+        if size == 0 || memory_size == 0 || offset > memory_size || (offset + size) > memory_size {
```

Comment:
- `size == 0` is an arbitrary restriction -- should we allow empty slices?
- `memory_size == 0` is a similarly arbitrary restriction, but at least this allows the nullptr check below to be a debug_assert, otherwise would need to check it

Reply:
Decided to allow empty slices.